spring-ai-core/src/main/java/org/springframework/ai/document/Document.java

```diff
@@ -25,130 +31,101 @@ public class Document {
 
 	// Type; introduce when support images, now only text.
 
-	private String text;
+	/**
+	 * Document content. TODO: To support binary content (image, audio ...)
+	 *
+	 * - One option is to change the content type to byte[]. Another option is to use
+	 * generics, e.g. Document<String> vs Document<byte[]>
+	 */
+	private String content;
+
+	/**
+	 * TODO: do we need the embedding field in the Document? Currently it is used only for
+	 * by the InMemoryVectorStore.
+	 *
+	 * Embedding of the document. Note: ephemeral field.
+	 */
+	@JsonProperty(index = 100)
+	private List<Double> embedding = new ArrayList<>();
+
+	/**
+	 * Mutable, ephemeral, content to text formatter. Defaults to Document text.
+	 */
+	@JsonIgnore
+	private ContentFormatter contentFormatter = DEFAULT_CONTENT_FORMATTER;
 
 	@JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
-	public Document(@JsonProperty("text") String text) {
-		this(text, new HashMap<>());
+	public Document(@JsonProperty("content") String content) {
```

Comment:
we will need to change the version of the project, might as well go  for 0.7.0 snapshots, as this is a breaking api change and I don't want to disrupt the azure/spring-ai workshops and other working examples.